bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts

```diff
@@ -221,19 +235,27 @@ export class OverviewComponent implements OnInit, OnDestroy {
   }
 
   private locateOrganizationsWithIncomingPaymentIssues(
+    organization: Organization,
     sub: OrganizationSubscriptionResponse,
     billing: BillingResponse,
-  ) {
+  ): FreeTrial {
     const trialEndDate = sub?.subscription?.trialEndDate;
-    this.isOwner = this.organization?.isOwner;
-    this.isTrialing = sub?.subscription?.status == "trialing";
-    this.defaultPaymentSource = billing;
+    const isOwner = organization?.isOwner;
+    const isTrialing = sub?.subscription?.status == "trialing";
+    const defaultPaymentSource = billing;
     if (trialEndDate) {
       const today = new Date();
       const trialEnd = new Date(trialEndDate);
       const timeDifference = trialEnd.getTime() - today.getTime();
       this.trialRemainingDays = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
     }
+    return {
+      trialing: isTrialing,
+      remainingDays: this.trialRemainingDays,
+      message: this.freeTrialMessage,
```

Comment:
You need to convert the getter `get freeTrialMessage() {` into normal function, since the computation of the message is done here and we should not rely on any of `this.` references, to reduce the amount of side effects.
```suggestion
      message: this.getFreeTrialMessage(trialRemainingDays),
```

and then the `trialRemainingDays` method itself:
```
  getFreeTrialMessage(trialRemainingDays: number) {
    if (trialRemainingDays >= 2) {
    ...
```